### PR TITLE
[Merged by Bors] - chore(HashMap): deprecate unused API additions

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Data/HashMap.lean
Comment (grunweg) [inline Mathlib/Data/HashMap.lean:32]: What do you think about: "This addition to mathlib is unused: if you need this, please file at issue in the `batteries` repository."?

**Review by grunweg — COMMENTED** [review]
One question, otherwise looks good. Feel free to `maintainer merge` on my behalf (and/or ask for a second opinion) when you considered it.

File: Mathlib/Data/HashMap.lean
Comment (semorrison) [reply Mathlib/Data/HashMap.lean]: Just noting that `Batteries.HashMap` is planned for wholesale deprecation shortly, but sure.

Comment (eric-wieser) [reply Mathlib/Data/HashMap.lean]: Is there a tracking issue / Zulip discussion for that?

Comment (semorrison) [reply Mathlib/Data/HashMap.lean]: Not that I'm aware of.

File: Mathlib/Data/HashMap.lean
Comment (adomani) [inline Mathlib/Data/HashMap.lean:33]:
```suggestion
  please file an issue in the Batteries repository." (since := "2024-06-12")]
```

File: Mathlib/Data/HashMap.lean
Comment (adomani) [inline Mathlib/Data/HashMap.lean:39]:
```suggestion
  please file an issue in the Batteries repository." (since := "2024-06-12")]
```

File: Mathlib/Data/HashMap.lean
Comment (adomani) [inline Mathlib/Data/HashMap.lean:45]:
```suggestion
  please file an issue in the Batteries repository." (since := "2024-06-12")]
```

File: Mathlib/Data/HashMap.lean
Comment (adomani) [inline Mathlib/Data/HashMap.lean:57]:
```suggestion
  please file an issue in the Batteries repository." (since := "2024-06-12")]
```

**Review by adomani — COMMENTED** [review]
I indeed never used these definitions!  : 😊

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Pull request closed without merging**
